src/app/sessions/book/page.tsx

```diff
@@ -32,8 +80,21 @@ export default function BookSessionPage() {
       if (response.ok) {
         // Navigate to confirmation page on success
         router.push("/sessions/book/confirmation");
+      } else if (response.status === 400) {
+        const responseData = await response.json();
+        if (responseData.message === "Game session at max capacity") {
+          const updatedCart = cart.filter((session) => !session.isFull);
+          useCartStore.getState().updateCart(updatedCart);
```

Comment:
does this work? can you define the updateCart function up above according to how these hooks are meant to be used. (see implementation above / in other areas.)